src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -202,6 +209,33 @@ static async ValueTask<QuicConnection> StartConnectAsync(QuicClientConnectionOpt
     /// </summary>
     public IPEndPoint LocalEndPoint => _localEndPoint;
 
+    /// <summary>
+    /// Returns the last known number of bidirectional stream that can be opened on this connection.
+    /// </summary>
+    public int AvailableBidirectionalStreamsCount => Volatile.Read(ref _availableBidirectionalStreamsCount);
+    /// <summary>
+    /// Returns the last known number of unidirectional stream that can be opened on this connection.
+    /// </summary>
+    public int AvailableUnidirectionalStreamsCount => Volatile.Read(ref _availableUnidirectionalStreamsCount);
+
+    /// <summary>
+    /// Occurres when an additional stream capacity has been released by the peer. Corresponds to receiving MAX_STREAMS frame.
+    /// </summary>
+    public event QuicConnectionStreamsAvailableEventHandler? StreamsAvailable;
+    private async void OnStreamsAvailable(int bidirectionalStreamsCountIncrement, int unidirectionalStreamsCountIncrement)
```

Comment:
```suggestion
    private async Task OnStreamsAvailable(int bidirectionalStreamsCountIncrement, int unidirectionalStreamsCountIncrement)
```

Reply:
Why?

Reply:
Gag reflex whenever I see `async void` 😃
E.g. it could be a regular `async Task` with the caller doing `_ = OnStreamsAvailable()`, like we do for `InjectNewHttp3ConnectionAsync`.

It's probably fine in this case though.

Reply:
I actually prefer `async void` to `_ = BlaBlaAsync()` :smile:  And we have a precedent for it in S.N.Quic (which I authored :smiley: )